wee_slack.py

```diff
@@ -2880,7 +2880,7 @@ def linkify_text(message, team, channel):
         .replace('>', '&gt;')
         .split(' '))
     for item in enumerate(message):
-        targets = re.match('^\s*([@#])([\w.-]+[\w. -])(\W*)', item[1])
+        targets = re.match('^\s*([@#])([\w\(.-]+[\w\). -])(\W*)', item[1], re.UNICODE)
```

Comment:
I tried to understand why this regex is written like this to figure out if your parenthesis are in the right places, but it doesn't make sense to me. Before your additions the two character classes (`[\w.-]` and `[\w. -]`) are equal except for a space. However, the message is split on spaces, so what is matched here will never contain spaces.

Either way, by placing the end parenthesis in the last class you define that nicks with parenthesis have to end with an end parenthesis. However, nicks in Slack are allowed to have parenthesis in the middle, so you should place it together with the start parenthesis. You can probably drop the last character class (`[\w. -]`), otherwise you will have to add both parenthesis in both classes.

Reply:
Ok, I changed the regexp as you suggested and wrote a test for that case as well.